src/task/MonitorTask.ts

```diff
@@ -296,11 +298,41 @@ export default class MonitorTask
             Maybe.of(observer.connectionDidBecomePoor).map(f => f.bind(observer)());
           });
         }
-      } else {
+      },
+      () => {
         this.context.audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
           Maybe.of(observer.connectionDidBecomeGood).map(f => f.bind(observer)());
         });
       }
+    );
+
+    this.enactHealthPolicy(
+      this.sendingAudioFailureHealthPolicy,
+      connectionHealthData,
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioFailed');
+      },
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioRecovered');
+      }
+    );
+  }
+
+  private enactHealthPolicy(
+    healthPolicy: BaseConnectionHealthPolicy,
```

Comment:
Since all health policies implement `ConnectionHealthPolicy` suggest this to type to `ConnectionHealthPolicy` (interface) and not `BaseConnectionHealthPolicy` (class).

Reply:
Sure I was hesitant because I needed to add a readonly property for capturing the name of the health policy for logging purpose and I was not sure if I can add that to the interface, but it doesn't seem to be an issue, will change